tools/tests/com.liferay.ide.gradle.core.tests/src/com/liferay/ide/gradle/core/tests/GradleParseTests.java

```diff
@@ -0,0 +1,181 @@
+
+        final File outputFile = new File( "projects/testParseOutput/testParse2.gradle" );
+
+        assertEquals( readFile( outputFile ), readFile( testfile ) );
+    }
+
+    @Test
+    public void addDependenceWithoutDendendenceBlock() throws IOException, RecognitionException, TokenStreamException
+    {
+        final File inputFile = new File( "projects/testParseInput/testParse3.gradle" );
+
+        GradleDependenceParser gradleScriptASTParser = new GradleDependenceParser( inputFile );
+
+        FindDependenceVisitor visitor = gradleScriptASTParser.walkScript(
+            "\tcompile group: \"com.liferay\", name:\"com.liferay.bookmarks.api\", version:\"1.0.0\"" );
+
+        int dependenceLineNum = visitor.getDependenceLineNum();
+
+        assertEquals( -1, dependenceLineNum );
+
+        Files.write( testfile.toPath(), gradleScriptASTParser.getScripts(), StandardCharsets.UTF_8 );
+
+        final File outputFile = new File( "projects/testParseOutput/testParse3.gradle" );
+
+        assertEquals( readFile( outputFile ), readFile( testfile ) );
+    }
+
+    @Test
+    public void addDependenceInSameLine() throws IOException, RecognitionException, TokenStreamException
+    {
+        final File inputFile = new File( "projects/testParseInput/testParse4.gradle" );
+
+        GradleDependenceParser gradleScriptASTParser = new GradleDependenceParser( inputFile );
+
+        FindDependenceVisitor visitor = gradleScriptASTParser.walkScript(
+            "\tcompile group: \"com.liferay\", name:\"com.liferay.bookmarks.api\", version:\"1.0.0\"" );
+
+        int dependenceLineNum = visitor.getDependenceLineNum();
+
+        assertEquals( 23, dependenceLineNum );
+
+        Files.write( testfile.toPath(), gradleScriptASTParser.getScripts(), StandardCharsets.UTF_8 );
+
+        final File outputFile = new File( "projects/testParseOutput/testParse4.gradle" );
+
+        assertEquals( readFile( outputFile ), readFile( testfile ) );
+    }
+
+    @Test
+    public void addDependenceInClosureLine() throws IOException, RecognitionException, TokenStreamException
+    {
+        final File inputFile = new File( "projects/testParseInput/testParse5.gradle" );
+
+        GradleDependenceParser gradleScriptASTParser = new GradleDependenceParser( inputFile );
+
+        FindDependenceVisitor visitor = gradleScriptASTParser.walkScript(
+            "\tcompile group: \"com.liferay\", name:\"com.liferay.bookmarks.api\", version:\"1.0.0\"" );
+
+        int dependenceLineNum = visitor.getDependenceLineNum();
+
+        assertEquals( 24, dependenceLineNum );
+
+        Files.write( testfile.toPath(), gradleScriptASTParser.getScripts(), StandardCharsets.UTF_8 );
+
+        final File outputFile = new File( "projects/testParseOutput/testParse5.gradle" );
+
+        assertEquals( readFile( outputFile ), readFile( testfile ) );
+    }
+
+    public String readFile( File file ) throws FileNotFoundException, IOException
```

Comment:
There is a method in CoreUtil for readin ga file to a String.  Use that one instead.